File: src/pocket_watch_ffi.erl
-module(pocket_watch_ffi).

-export([elapsed/1, monotonic_now/0]).

elapsed(Fun) ->
    Start = erlang:monotonic_time(),
    Return = Fun(),
    Stop = erlang:monotonic_time(),
    Elapsed = float(erlang:convert_time_unit(Stop - Start, native, nanosecond)),

    {Return, Elapsed}.

monotonic_now() ->
    float(erlang:convert_time_unit(erlang:monotonic_time(), native, nanosecond)).

File: src/pocket_watch_ffi.mjs
export function elapsed(fun) {
  const start = globalThis.performance.now();
  const ret = fun();
  const stop = globalThis.performance.now();

  return [ret, (stop - start) * 1000000];
}

export function monotonic_now() {
  return globalThis.performance.now() * 1000000;
}
